lua/bacon/init.lua

```diff
@@ -268,5 +270,57 @@ function Bacon.bacon_next()
 	end
 end
 
+local function find_loc_file()
+	local dir = ""
+	repeat
+		local file = dir .. ".bacon-locations"
+		if file_exists(file) then
+			return file
+		end
+		dir = "../" .. dir
+	until not file_exists(dir)
+end
+
+local function watch_loc_file(path)
+	-- Adapted from https://github.com/rktjmp/fwatch.nvim
+
+	local handle = loop.new_fs_event()
+
+	local flags = {
+		watch_entry = false,
+		stat = false,
+		recursive = false,
+	}
+
+	local event_cb = function(err, _, _)
+		if err then
+			error("Autoload failed to watch .bacon-locations")
+		else
+			vim.schedule(function()
+				Bacon.bacon_load()
+			end)
+		end
+	end
+	loop.fs_event_start(handle, path, flags, event_cb)
+
+	return handle
+end
+
+if config.options.autoload then
```

Comment:
I don't think the "options" is right here, it should be `config.autoload` with the config in exemple